rpcserver.go

```diff
@@ -809,15 +814,151 @@ func (r *rpcServer) VerifyProof(ctx context.Context,
-		Valid: valid,
+	return &tarorpc.VerifyProofResponse{
+		Valid:        valid,
+		DecodedProof: rpcProof,
 	}, nil
 }
 
+// DecodeProof attempts to decode a given proof file that claims to be anchored
+// at the specified genesis point.
+func (r *rpcServer) DecodeProof(ctx context.Context,
+	in *tarorpc.DecodeProofRequest) (*tarorpc.DecodeProofResponse, error) {
+
+	if len(in.RawProof) == 0 {
+		return nil, fmt.Errorf("proof file must be specified")
+	}
+
+	var proofFile proof.File
+	err := proofFile.Decode(bytes.NewReader(in.RawProof))
+	if err != nil {
+		return nil, fmt.Errorf("unable to decode proof file: %w", err)
+	}
+
+	resp, err := marshalProofFile(proofFile, in.ProofIndex)
+	if err != nil {
+		return nil, fmt.Errorf("unable to marshal proof: %w", err)
+	}
+
+	return resp, nil
+}
+
+// marshalProofFile turns a proof file into an RPC DecodeProofResponse.
+func marshalProofFile(proofFile proof.File, index int64) (*tarorpc.DecodeProofResponse, error) {
+	var (
+		decodedProof *proof.Proof
+		err          error
+	)
+
+	if index < 0 {
+		decodedProof, err = proofFile.LastProof()
+		if err != nil {
+			return nil, err
+		}
+		index = int64(proofFile.NumProofs() - 1)
+	} else {
+		decodedProof, err = proofFile.ProofAt(uint32(index))
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	var (
+		a                 = decodedProof.Asset
+		rpcAsset          = &tarorpc.Asset{}
+		assetID           = a.Genesis.ID()
+		anchorTxId        = decodedProof.AnchorTx.TxHash().String()
+		anchorOutputIndex = strconv.FormatUint(uint64(decodedProof.InclusionProof.OutputIndex), 10)
+		anchorOutpoint    = anchorTxId + ":" + anchorOutputIndex
+	)
+
+	var anchorTxBuf bytes.Buffer
+	err = decodedProof.AnchorTx.Serialize(&anchorTxBuf)
+	if err != nil {
+		return nil, fmt.Errorf("unable to serialize "+
+			"anchor tx: %w", err)
+	}
+	anchorTxBytes := anchorTxBuf.Bytes()
+
+	var txMerkleProofBuf bytes.Buffer
+	if err := decodedProof.TxMerkleProof.Encode(&txMerkleProofBuf); err != nil {
+		return nil, fmt.Errorf("unable to encode serialized Bitcoin merkle proof: %w",
+			err)
+	}
+	txMerkleProofBytes := txMerkleProofBuf.Bytes()
+
+	var inclusionProofBuf bytes.Buffer
+	if err := decodedProof.InclusionProof.Encode(&inclusionProofBuf); err != nil {
+		return nil, fmt.Errorf("unable to encode inclusion proof: %w",
+			err)
+	}
+	inclusionProofBytes := inclusionProofBuf.Bytes()
+
+	var exclusionProofs [][]byte
+	for _, exclusionProof := range decodedProof.ExclusionProofs {
+		var exclusionProofBuf bytes.Buffer
+		if err := exclusionProof.Encode(&exclusionProofBuf); err != nil {
+			return nil, fmt.Errorf("unable to encode exclusion proofs: %w",
+				err)
+		}
+		exclusionProofBytes := exclusionProofBuf.Bytes()
+
+		exclusionProofs = append(exclusionProofs, exclusionProofBytes)
+	}
+
+	var bootstrapInfoBuf bytes.Buffer
+	if err := a.Genesis.Encode(&bootstrapInfoBuf); err != nil {
+		return nil, fmt.Errorf("unable to encode genesis: %w",
+			err)
+	}
+	rpcAsset = &tarorpc.Asset{
```

Comment:
can [`marshalAsset`](https://github.com/lightninglabs/taro/blob/main/rpcserver.go#L406) help here?

if we need a subset we can create a `marshalHumanReadableAsset` helper